packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -94,7 +94,12 @@ function HeaderToolbar() {
 					'Generic label for block inserter button'
 				) }
 			/>
-			{ isLargeViewport && <ToolbarItem as={ ToolSelector } /> }
+			{ isLargeViewport && (
+				<ToolbarItem
+					as={ ToolSelector }
+					disabled={ ! isTextModeEnabled }
```

Comment:
This check should be inverted.
```suggestion
					disabled={ isTextModeEnabled }
```